backend/wmg/api/v1.py

```diff
@@ -128,7 +130,7 @@ def build_expression_summary(query_result: DataFrame) -> dict:
     return structured_result
 
 
-def build_dot_plot_matrix(query_result: DataFrame, cell_counts: DataFrame) -> DataFrame:
+def build_dot_plot_matrix(query_result: DataFrame, cell_counts: DataFrame) -> Tuple[DataFrame, DataFrame]:
```

Comment:
Can you update this function name and description to match the added functionality?
or even better refactor to  create a new function that calls `build_dot_plot_matrix` and a separate function calling fetching the cell counts